pkg/cri/server/container_stop.go

```diff
@@ -129,9 +130,13 @@ func (c *criService) stopContainer(ctx context.Context, container containerstore
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse stop signal %q", stopSignal)
 		}
-		log.G(ctx).Infof("Stop container %q with signal %v", id, sig)
-		if err = task.Kill(ctx, sig); err != nil && !errdefs.IsNotFound(err) {
-			return errors.Wrapf(err, "failed to stop container %q", id)
+
+		if container.IsSignaled == nil ||
+			(container.IsSignaled != nil && atomic.CompareAndSwapUint32(container.IsSignaled, 0, 1)) {
```

Comment:
for the else case (where is signaled was already set) please output log.G(ctx).Infof("Skipping the sending of signal %v to container %q because a prior stop with timeout>0 request already sent the signal", sig, id)